--- a/src/CMakeLists.txt	2026-03-19 14:08:37.823737773 -0400
+++ b/src/CMakeLists.txt	2026-03-19 16:23:59.993885166 -0400
@@ -390,13 +390,21 @@
     IMPORTED_LINK_INTERFACE_LANGUAGES "CXX")
 endif(WITH_BREAKPAD)
 
+option(WITH_SYSTEM_OPENTELEMETRY "build against system opentelemetry" OFF)
 if(WITH_JAEGER)
-  find_package(thrift 0.13.0 REQUIRED)
-  include(BuildOpentelemetry)
-  build_opentelemetry()
   add_library(jaeger_base INTERFACE)
-  target_link_libraries(jaeger_base INTERFACE opentelemetry::libopentelemetry
-    thrift::libthrift)
+  if(WITH_SYSTEM_OPENTELEMETRY)
+    find_package(opentelemetry-cpp REQUIRED)
+    target_link_libraries(jaeger_base INTERFACE
+      opentelemetry-cpp::trace
+      opentelemetry-cpp::resources
+      opentelemetry-cpp::otlp_http_exporter
+      opentelemetry-cpp::common)
+  else()
+    include(BuildOpentelemetry)
+    build_opentelemetry()
+    target_link_libraries(jaeger_base INTERFACE opentelemetry::libopentelemetry)
+  endif()
 endif()
 
 set(mds_files)
